### DISCUSSION TITLE
Java Language Server

### DISCUSSION THREAD

**Discussion by philiprbrenan** [geany/geany#3286] [Ideas]
Might it be possible to have Geany communicate with  

https://github.com/eclipse/eclipse.jdt.ls

to get accurate suggestions for methods that can be called after a reference or class name?  

I start a lot of new programmers off with Geany + Java so this would be a powerful combination.  At the moment they tend to use VS by default : I would like to be able to offer them this feature as a compelling reason to use Geany instead of VS because Geany is much better than VS except in this one area.

**Comment by elextr** [comment]
Using a C/C++ language server was once tried, but it was abandoned because of the magnitude of the changes needed to Geany.

Basically Geany is written to do everything itself and would need many changes to switch between its existing behaviour for languages without servers and using a server for languages that have one.
